lib/spack/spack/spec.py

```diff
@@ -2893,6 +2893,36 @@ def ensure_valid_variants(spec):
         if not_existing:
             raise vt.UnknownVariantError(spec, not_existing)
 
+    def add_variant_values(self, variant_name, values):
+        """If it is not already there, adds the variant named
+        `variant_name` to the spec `spec` based on the definition
+        contained in the package metadata. Used to add values
+        to a variant without being sensitive to the variant being
+        single or multi-valued. If the variant already exists
+        on the spec it is assumed to be multi-valued and
+        the values are appended.
+
+        Args:
+           variant_name: the name of the variant to add or append to
+           values: the value or values (as a tuple) to add/append
+                   to the variant
+        """
```

Comment:
Minor. I'd suggest a renaming and reformatting of the docstring:
```suggestion
    def update_variant(self, variant_name, values):
        """If it is not already there, adds the variant named `variant_name` 
        to the spec `spec` based on the definition contained in the package 
        metadata. 
        
        Used to add values to a variant without being sensitive to the variant
        being single or multi-valued. If the variant already exists on the 
        spec it is assumed to be multi-valued and the values are appended.

        Args:
            variant_name (str): the name of the variant to add or append to
            values (str or tuple): the value or values (as a tuple) to 
                add/append to the variant
        """
```
Let me know what you think of that.

Reply:
I took it a few steps further, especially regarding making the validation step clear.